Add introduction of the web report to the docs

zinwang opened the pull request:
**Description**

This PR aims to close Issue #400. 

**Document Changes**
+ Add a new content, Report. 
+ Add introduction of the Web Report to Report.

zinwang commented:
Here is the content of report.rst.

Report
===


Web Report
------------------------
You can analyze an APK sample and produce a beautiful summary report in HTML format with the following command.

    quark -a Ahmyth.apk -s -w test.html 

the meanings of the options:

* -a for sample file input
* -s for producing the summary report
* -w for Web Report file output

Here is a [demo](https://pulorsok.github.io/ruleviewer/web-report-demo) of the Web Report. 
And the page can be divided into four parts:

* Analysis Result
* Sample Information
* Radar Chart
* Detected Crimes

![](https://camo.githubusercontent.com/9e25807aa6c0173b995dd94f867c7eae461e29a61537d1791c3633d33e913041/68747470733a2f2f692e696d6775722e636f6d2f684733416738742e706e67)



Analysis Result
==========================
First, Analysis Result gives you the statics of the crimes detected by Quark.

You can see the following:

1. There is a doughnut chart on the left, showing the ratio of the crimes with 100% confidence in all crimes detected. 

2. On the right of the doughnut chart, you can see the number of crimes in different confidence levels. The confidence levels are 0%, 20%, 40%, 60%, 80% and 100%. The higher the level is, the more we are sure the behavior is malicious. 

![](https://i.imgur.com/XkkaCvJ.png)

Sample Information
==========================
This part shows the basic information of the sample, including file name, MD5 hash value, file size, and labels of detected rules with 100% confidence. 


![](https://i.imgur.com/7hSXJDZ.png)



Radar Chart
==========================
In the part of the Radar Chart, 

1. You can easily choose the labels you want on the right. And they will be the dimensions to analyze in the radar chart.

2. Then, the page will draw the chart on the left. And the values on the axes are the confidences of crimes corresponding to the labels. 

3. If you decide to draw the chart all over again, you can use the deselect button to uncheck all the labels.

4. Also, the labels of the crimes detected with 100% confidence are displayed.

![](https://i.imgur.com/TDiadQZ.png)



Detected Crimes
==========================

In this part,

1. It shows the rule numbers, crime descriptions, and the confidence of the crimes. 

![](https://i.imgur.com/h1Ai1VZ.png)


2. You can find crimes by searching specific strings in the Crime Description with the searching field. 

3. And you can show crimes with particular confidence using the drop-down list.


![](https://i.imgur.com/n9Y3uKx.png)

haeter525 commented:
@18z Please review the document about the web report. Thank you!

18z commented:
Thank you @zinwang  and @haeter525 !

I think we should ensure that in every picture we provide, the content must be clean and tidy.

Two suggestions:
1. One picture one story. Make the picture has only one story and nothing but the story.
For example

Instead of 
![sample_info](https://camo.githubusercontent.com/019a150018afaacf06a77979352536d6fe88bec9ed83e9bd41e8e644eaf7d416/68747470733a2f2f692e696d6775722e636f6d2f376853584a445a2e706e67)

Use
![Screen Shot 2022-10-21 at 9 34 05 AM](https://user-images.githubusercontent.com/593954/197090587-81949fa8-5512-460c-aa9c-651424c698fa.png)

2. Make the content centered, and the margins consistent.

zinwang commented:
@18z 
I have fixed the problem in the images and updated the preview above. Please have a look.
As for the typo and the light grid lines in the Radar Chart section,
I have created issue #405 to fix them.

18z commented:
hi @zinwang 

I have a new idea. Let's just keep the picture of the whole report and remove all the others.
Because I think the snapshots are difficult to present nicely.

At the top of the document, we show the report directly. 
Then, we describe each part of the report accordingly. I think this should be clear enough.

For example:

![68747470733a2f2f692e696d6775722e636f6d2f684733416738742e706e67](https://user-images.githubusercontent.com/593954/197925513-f4a777d2-b512-4969-89c5-2725e2341a85.png)

### Analysis Result
Description

### Sample Information
Description

### Radar Chart
Description

### Label selection
Description

### 100% Confidence Crimes
Description

### Search Crime
Description

### Confidence Filter
Description

### List of Rules and Crime
Description

zinwang commented:
@18z 
Here is the modified version following the advice you gave above.
Please have a look. Thanks!

Report
===


Web Report
---
You can analyze an APK sample and produce a beautiful summary report in HTML format through the following command.

```
quark -a Ahmyth.apk -s -w test.html 
```

The meanings of the options:

* -a, APK sample file input
* -s, Show summary report
* -w, Generate Web Report output

Here is a [demo](https://pulorsok.github.io/ruleviewer/web-report-demo) of the Web Report. The image below demonstrates the overview of the Web Report. 

![](https://camo.githubusercontent.com/9e25807aa6c0173b995dd94f867c7eae461e29a61537d1791c3633d33e913041/68747470733a2f2f692e696d6775722e636f6d2f684733416738742e706e67)


Analysis Result
---
First, at the top-left corner, there is an Analysis Result section. It gives you the statistics of the crimes detected by Quark.

It contains two parts in this section:

#### Doughnut Chart
  A doughnut chart shows the ratio of the crimes with 100% confidence in all crimes detected. 

#### Statistics
  Statistics show crimes in different confidence levels from 0% to 100%. The higher the level is, the more we are sure the behavior is malicious.

    
Sample Information
---
At the top-right corner, Sample Information shows the basic information of the sample, including the file name, MD5 hash value, file size, and the labels of detected rules with 100% confidence. 



Radar Chart
---
In this section, you can generate a custom radar chart using labels from detected rules. This way, you can compare the confidence level of different dimensions in the sample in an easy-to-understand form. 

1. First, you can choose the labels you want as the dimensions to analyze in the label section on the right.

2. Then, the page will plot the radar chart on the left according to your selection. In the chart, the values on the axes are the confidences of crimes corresponding to the labels.

3. If you want to replot the chart, you can use the "unselect button" to uncheck all the labels.

4. This section also lists the labels of the detected crimes that have 100% confidence.


Search Crimes
---
You can use this search field to search crimes with specific strings in crime descriptions. It helps you find particular crimes quickly.


Confidence Filter
---
With this filter, you can view crimes at specific confidence levels.


Detected Crimes
---

At the bottom of this report, it shows the rule numbers, crime descriptions, and the confidence level of the crimes detected. With this information, you can know what the sample does and how malicious the behaviors are.
